x/qgb/orchestrator/parser.go

```diff
@@ -0,0 +1,96 @@
+package orchestrator
+
+import (
+	"fmt"
+
+	"github.com/celestiaorg/celestia-app/app"
+	qgbtypes "github.com/celestiaorg/celestia-app/x/qgb/types"
+	sdkcodec "github.com/cosmos/cosmos-sdk/codec"
+	sdkcodectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/std"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/gogo/protobuf/proto"
+	coretypes "github.com/tendermint/tendermint/types"
+)
+
+var _ QGBParserI = QGBParser{}
+
+type QGBParserI interface {
```

Comment:
do we need an interface here? are we planning for multiple different implementations?

Reply:
I added an interface in case we need to mock this parser in unit tests. For example, if we want to test the ingestion service with random data without worrying if it is random

Reply:
I see. For me, I was thinking more of the ingestion service tests. Since it will need to extract data from a node (or a mock of a node that will return random data) and will need to index that data into the right place and be able to load it correctly.
What do you think about this 
 testing scenario?

Reply:
How will it index random data? Unless its hashing the random data, won't it need some metadata to use as an index? I could be missing something, but it doesn't seem that difficult to mock create mock data. When we used mocks before, it was because it was very difficult to spin up an entire chain to do unit tests, so the mocks make sense. I could be wrong tho, and if it is difficult, then we should keep it.

Reply:
I will add a TODO, and when implementing tests, we see if it makes sense to keep it or not. How does that sound?

Reply:
sure, that works

Reply:
[68aad0d](https://github.com/celestiaorg/celestia-app/pull/878/commits/68aad0d9d7fecfbcc23ccb6971bf8fafbe1f54f4)